services/src/modules/authentication/jwt-decoder-plugin.ts

```diff
@@ -0,0 +1,43 @@
+import * as http from 'http';
+import { decode } from 'jsonwebtoken';
+import { FastifyRequest, FastifyInstance } from 'fastify';
+import * as fp from 'fastify-plugin';
+import logger from '../logger';
+
+type DecodedJWT = {
+  header: Record<string, unknown>;
+  payload: Record<string, unknown>;
+  signature: string;
+};
+
+declare module 'fastify' {
+  interface FastifyRequest<
+    HttpRequest = http.IncomingMessage,
+    Query = DefaultQuery,
+    Params = DefaultParams,
+    Headers = DefaultHeaders,
+    Body = DefaultBody
+  > {
+    decodeJWT(): Promise<void>;
+    decodedJWT?: DecodedJWT | null;
+  }
+}
+
+async function decodeRequestJwt(this: FastifyRequest) {
```

Comment:
It is very unusual to name an argument `this`, you should change it to something else

Reply:
This is not an argument.
See [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions).